src/index.js

```diff
@@ -537,14 +537,27 @@ const useLocation = () => {
 
 const useNavigate = () => {
   const context = useContext(LocationContext);
+  const baseContext = useContext(BaseContext);
 
-  if (!context) {
+  if (!context || !baseContext) {
     throw new Error(
       "useNavigate hook was used but a LocationContext.Provider was not found in the parent tree. Make sure this is used in a component that is a child of Router"
     );
   }
 
-  return context.navigate;
+  const { location, navigate } = context;
+
+  const results = match(baseContext.basepath, location.pathname);
```

Comment:
this is essentially `useMatch`, but that requires an argument which I don't have until `navigate` is called